src/test/java/plugins/GitLabPluginTest.java

```diff
@@ -0,0 +1,197 @@
+        tk.setToken(token);
+        tk.setId(id);
+        tk.create();
+    }
+
+    public void configureGitLabServer() throws IOException {
+        jenkins.configure();
+
+        // server configuration
+        GitLabServerConfig serverConfig = new GitLabServerConfig(jenkins);
+        serverConfig.configureServer(container.getHttpUrl().toString());
+        jenkins.save();
+    }
+
+    @Test
+    public void testGitLabMultibranchPipeline() throws IOException, GitLabApiException {
+        createGitLabToken(privateTokenAdmin, "GitLab Personal Access Token");
+        configureGitLabServer();
+        createRepo();
+        container.createBranch(getPrivateTokenAdmin(), repoName);
+
+        final WorkflowMultiBranchJob multibranchJob = jenkins.jobs.create(WorkflowMultiBranchJob.class);
+        this.configureJobWithGitLabBranchSource(multibranchJob, adminUserName, repoName);
+        multibranchJob.save();
+
+        multibranchJob.waitForBranchIndexingFinished((int)time.seconds(20));
+        this.assertBranchIndexing(multibranchJob);
+
+        final WorkflowJob successJob1 = multibranchJob.getJob("main");
+        final WorkflowJob successJob2 = multibranchJob.getJob("firstbranch");
+        final WorkflowJob successJob3 = multibranchJob.getJob("MR-1");
+        final WorkflowJob failureJob = multibranchJob.getJob("failedjob");
+
+        this.assertExistAndResult(successJob1, true);
+        this.assertExistAndResult(successJob2, true);
+        this.assertExistAndResult(successJob3, true);
+        this.assertExistAndResult(failureJob, false);
+
+        // delete the repo when finished
+        container.deleteRepo(getPrivateTokenAdmin(), repoName);
+    }
+
+    private void assertBranchIndexing(final WorkflowMultiBranchJob job) {
+        assertThat(driver, hasContent("Scan GitLab Project Now"));
+        final String branchIndexingLog = job.getBranchIndexingLog();
+
+        assertThat(branchIndexingLog, containsString("Scheduled build for branch: main"));
+        assertThat(branchIndexingLog, containsString("Scheduled build for branch: firstbranch"));
+        assertThat(branchIndexingLog, containsString("Scheduled build for branch: failedjob"));
+        assertThat(branchIndexingLog, containsString("Scheduled build for branch: MR-1"));
+        assertThat(branchIndexingLog, containsString("1 merge requests were processed"));
+    }
+
+    private void configureJobWithGitLabBranchSource(final WorkflowMultiBranchJob job, String adminUserName, String project) {
+        final GitLabBranchSource ghBranchSource = job.addBranchSource(GitLabBranchSource.class);
+        ghBranchSource.setOwner(adminUserName);
+        ghBranchSource.setProject(adminUserName, project);
+    }
+
+    private void assertExistAndResult(final WorkflowJob job, final boolean withSuccess) {
+        final Build.Result expectedResult = (withSuccess) ? Build.Result.SUCCESS : Build.Result.FAILURE;
```

Comment:
I think you need to wait for the build to have finished here don't you?  it may be in progress or in the queue having not yet started?